st2actions/st2actions/scheduler.py

```diff
@@ -56,26 +58,42 @@ def process(self, request):
             LOG.exception('Failed to find liveaction %s in the database.', request.id)
             raise
 
+        # Apply policies defined for the action.
+        for policy_db in Policy.query(resource_ref=liveaction_db.action):
+            policy = policies.get_driver(policy_db.ref,
+                                         policy_db.policy_type,
+                                         **policy_db.parameters)
+            liveaction_db = policy.apply(liveaction_db)
```

Comment:
PolicyApplier shouldn't mutate the liveaction_db IMO. In other words, I am not too happy with policy.apply() returning a mutated liveaction object. I guess we have to decide if applying policy on action just means whether the action should be run or not.